pkg/cli/debug/debug.go

```diff
@@ -978,7 +977,7 @@ func (o *DebugOptions) getContainerCommand() []string {
 	if len(o.Command) == 1 && o.Command[0] == commandLinuxShell {
 		pod := o.Attach.Pod
 
-		if val, ok := pod.Spec.NodeSelector[kubeOSNodeSelector]; ok && val == "windows" {
+		if pod.Spec.OS != nil && pod.Spec.OS.Name == corev1.Windows {
```

Comment:
What is the reason of the change in this line?

Reply:
The reason is that `pod.Spec.NodeSelector[kubeOSNodeSelector]` is not set on the debug pod, making this if statement never evaluate to true.
It seems that the recommended way to check if a pod is meant for Windows is to use OS field, so I opted for that method.